### ISSUE TITLE
Sync Failure

### ISSUE THREAD

**Issue by tastrax** [lutraconsulting/qgis-mergin-plugin#286]
Traceback (most recent call last):

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\merginproject.py", line 437, in apply_pull_changes
    self.geodiff.create_changeset(basefile, src, server_diff)

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\deps\pygeodiff\main.py", line 65, in create_changeset
    return self.clib.create_changeset(base, modified, changeset)

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\deps\pygeodiff\geodifflib.py", line 200, in create_changeset
    _parse_return_code(res, "createChangeset")

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\deps\pygeodiff\geodifflib.py", line 41, in _parse_return_code
    raise GeoDiffLibError(msg)

Mergin.mergin.deps.pygeodiff.geodifflib.GeoDiffLibError: createChangeset

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\sync_dialog.py", line 306, in pull_timer_tick
    self.pull_conflicts = pull_project_finalize(self.job)

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\client_pull.py", line 555, in pull_project_finalize
    conflicts = job.mp.apply_pull_changes(job.pull_changes, job.temp_dir)

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\merginproject.py", line 445, in apply_pull_changes
    self.geodiff.make_copy_sqlite(f_conflict_file, dest)

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\deps\pygeodiff\main.py", line 221, in make_copy_sqlite
    return self.clib.make_copy_sqlite(src, dst)

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\deps\pygeodiff\geodifflib.py", line 328, in make_copy_sqlite
    _parse_return_code(res, "make_copy_sqlite")

  File "C:/Users/Owner/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\Mergin\mergin\deps\pygeodiff\geodifflib.py", line 41, in _parse_return_code
    raise GeoDiffLibError(msg)

Mergin.mergin.deps.pygeodiff.geodifflib.GeoDiffLibError: make_copy_sqlite

**Comment by wonder-sk** [comment]
hi @tastrax do you have some more context about how this happened? Does it happen regularly, or only once? Are you able to replicate the issue?

**Comment by tastrax** [comment]
Alas no - just went to sync and got the error. I had made a few syncs previously in the day (making form changes) and then up popped this error

**Comment by tastrax** [comment]
[client-log.txt](https://github.com/lutraconsulting/qgis-mergin-plugin/files/7142585/client-log.txt)

**Comment by saberraz** [comment]
@tastrax you seem to be using an old version of the plugin? Could you upgrade and see if it happens.

**Comment by tastrax** [comment]
Alas - same issue

![SyncFail](https://user-images.githubusercontent.com/8775395/132830231-71f0fafe-89e8-4a9a-b96b-5800c076ea62.JPG)

**Comment by wonder-sk** [comment]
It looks like the issue is that in Mergin plugin version 2021.4, the project directory got slightly broken (actually the internal `.mergin` sub-directory), and that's causing further errors even after upgrade to 2021.4.1. The solution at this point should be to delete the project directory and re-download the Mergin project - after that, you should stop getting these errors. We will try to issue a fix so that even broken project directory from older version of the plugin gets fixed automatically...

**Comment by wonder-sk** [comment]
Duplicate of #284
